tasks/notify.py

```diff
@@ -459,40 +460,58 @@ def send_notification(ctx: Context, alert_jobs):
 
 
 @task
-def send_failure_summary_notification(_, jobs: dict[str, any] | None = None, list_max_len=10):
+def send_failure_summary_notification(
+    _, jobs: dict[str, any] | None = None, list_max_len=10, jobowners=".gitlab/JOBOWNERS"
+):
+    from slack_sdk import WebClient
+
     if jobs is None:
         jobs = os.environ["JOB_FAILURES"]
         jobs = json.loads(jobs)
 
     # List of (job_name, (failure_count, total_count)) ordered by failure_count
     stats = sorted(
-        ((name, (fail, success)) for (name, (fail, success)) in jobs.items() if fail > 0),
+        ((name, (fail, total)) for (name, (fail, total)) in jobs.items() if fail > 0),
         key=lambda x: (x[1][0], x[1][1] if x[1][1] is not None else 0),
         reverse=True,
-    )[:list_max_len]
+    )
 
     # Don't send message if no failure
     if len(stats) == 0:
         return
 
-    # Create message
-    message = ['*Daily Job Failure Report*']
-    message.append('These jobs had the most failures in the last 24 hours:')
-    for name, (fail, total) in stats:
-        link = get_ci_visibility_job_url(name)
-        message.append(f"- <{link}|{name}>: *{fail} failures*{f' / {total} runs' if total else ''}")
+    # Partitionate by channels as some teams share the same slack channel (avoid duplicate messages)
+    partition = partitionate([name for name, _ in stats], jobowners, get_channels=True)
+
+    # team_stats[team] = [(job_name, (failure_count, total_count)), ...]
+    team_stats = {}
+    for channel in partition:
+        team_stats[channel] = [(name, stat) for (name, stat) in stats if name in partition[channel]]
+        team_stats[channel] = team_stats[channel][:list_max_len]
+
+    def send_summary(channel, stats):
+        # Create message
+        message = ['*Daily Job Failure Report*']
+        message.append('These jobs you own had the most failures in the last 24 hours:')
+        for name, (fail, total) in stats:
+            link = CI_VISIBILITY_JOB_URL.format(quote(name))
+            message.append(f"- <{link}|{name}>: *{fail} failures*{f' / {total} runs' if total else ''}")
+
+        message.append(
+            'Click <https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Ajob%20env%3Aprod%20%40git.repository.id%3A%22gitlab.ddbuild.io%2FDataDog%2Fdatadog-agent%22%20%40ci.pipeline.name%3A%22DataDog%2Fdatadog-agent%22%20%40ci.provider.instance%3Agitlab-ci%20%40git.branch%3Amain%20%40ci.status%3Aerror&agg_m=count&agg_m_source=base&agg_q=%40ci.job.name&agg_q_source=base&agg_t=count&fromUser=false&index=cipipeline&sort_m=count&sort_m_source=base&sort_t=count&top_n=25&top_o=top&viz=toplist&x_missing=true&paused=false|here> for more details.'
```

Comment:
This (and the other CI Visibility link) are a bit strange. If I understand this correctly, this goes to the CI Visibility page listing all job executions for the last week. Wouldn't it make more sense to have this scoped to the timeframe that was used to compute the data?